### PR TITLE
Fixes #221

### LINKED ISSUE #221: Multiple divisions are not RegExp
`Math.floor x / y / 2` parses as a regular expression instead of two divisions.

I think usually a space after `/` means "not a regular expression".

### PR DESCRIPTION
**Pull request by STRd6**

### CODE REVIEW

**Review by edemaine — CHANGES_REQUESTED** [review]
I find this a pretty obscure way to fix the issue.  I'd rather see `RegularExpressionBody` follow https://262.ecma-international.org/#prod-RegularExpressionFirstChar which it seems to be trying to do; I think space just needs to be in the `![*\/\r\n]` negative lookahead?

**Comment by STRd6** [comment]
The ambiguity comes from the interaction with the implicit function application since:

```coffee
x / 7 / 2
```

Get's interpreted as an implicit function call with a RegExp as the first argument (explicitly calling itself with 2 as an arg, a possible separate issue :P ).

This is valid JS:

```javascript
x = / /
```

The `RegularExpressionFirstChar` from the spec disallows line breaks (we probably should too but it's a also a different issue than this one).

**Review by edemaine — APPROVED** [review]
Ah, I understand now. `/` is a special character in that it can be a binary operator (verb) *and* the beginning of a literal (noun).

It looks like `//` isn't a valid regular expression, and non-heregexes can't have newlines, so I think I now believe this is the correct exclusion.

Sorry for the confusion and thanks for the explanation!

File: source/parser.hera [hunk 1]
```diff
@@ -547,6 +547,8 @@ NonNullAssertion
 # ie: the 'of' in 'for x of ...'
 AdditionalReservedWords
```
Comment (edemaine) [inline source/parser.hera:548]: I do wonder whether there's a better name than `AdditionalReservedWords`, like `ForbiddenImplicitCalls` or something. There's no reference to the other `ReservedWord` rule so it's not really additional.

Comment (STRd6) [reply source/parser.hera:548]: I'll update the name.

**Comment by STRd6** [comment]
No worries, the JS spec is vast and there is a lot to keep track of.

**Pull request merged**